### DISCUSSION TITLE
Drupal local site DNS for backstopJS docker container

### DISCUSSION THREAD

**Discussion by wilburec** [docksal/docksal#1730] [General]
We are using BackstopJS Visual Regression on our Drupal 9 websites, running our local installs on Docksal.  We create reference images from our LIVE site using the local Backstop command:
`backstop reference --docker` 
No troubles, the docker image can find the publicly available URL.

BUT when we run the test images using this command:
`backstop test --docker`
The docker browser cannot find the local url:
```
Puppeteer encountered an error while running scenario "Homepage"
Error: net::ERR_NAME_NOT_RESOLVED at http://wilder.docksal.site/
```

How do we get the backstop docker instance to 'see' the local url?

**Comment by lpeabody** [comment]
What docker commands are you issuing in order to execute the backstop tests? Are you using the backstop docker image? If it's through the image I've had to do something like this in the past:

```
  fin docker run -it --rm \
    -v $PROJECT_ROOT/tests/backstop:/src \
    --dns=$DOCKSAL_DNS1 --dns=$DOCKSAL_DNS2 \
    --env=SITE --env=VIRTUAL_HOST \
    --user $(id -u):$(id -g) \
    backstopjs/backstopjs:$BACKSTOP_VERSION "$@"
```

Based on what you're saying, the key part of that command is providing the `--dns` options with the Docksal DNS values.

**Comment by wilburec** [comment]
Thanks for your suggestion!  Based on that information, we are able to create an alias to execute backstop commands in the docksal container.

We are not running docksal_dns, so instead we manually add the host mapping as part of the command.  
```
docker run --rm -v $(pwd):/src --add-host="mysite.docksal.site:192.168.64.100" backstopjs/backstopjs "$@"
```
To automate this command for any environment, we created a command that works specifically in our backstop install directory - `mysite/tests/backstop`.  We created a bash alias to that used the directory name to generate the DNS entry:
```
alias bkstp='docker run --rm -v $(pwd):/src --add-host="$(basename ${PWD%/*/*}).docksal.site:192.168.64.100" backstopjs/backstopjs "$@"'
```
This can be used in the backstop testing folder to set the correct host entry.
